pytest/tools/prober/prober_split.py

```diff
@@ -106,21 +106,43 @@ def check_chunks(legacy_url: str, split_url: str, block):
             sys.exit(1)
 
 
+# Query gcp for the archive nodes, pick a random one and return its url.
+def get_random_legacy_url(chain_id):
+    cmd = [
+        'gcloud',
+        'compute',
+        'instances',
+        'list',
+    ]
+    logger.info(" ".join(cmd))
+    result = subprocess.run(cmd, text=True, capture_output=True)
+    stdout = result.stdout
+    lines = stdout.split('\n')
+    pattern = f'{chain_id}-rpc-archive-public'
```

Comment:
I failed to make the --filter argument of gcloud work properly so doing it in python instead.